libcnb-cargo/src/lib.rs

```diff
@@ -174,9 +77,27 @@ pub fn assemble_buildpack_directory(
     let bin_path = destination_path.as_ref().join("bin");
     fs::create_dir_all(&bin_path)?;
 
-    fs::copy(buildpack_binary_path.as_ref(), bin_path.join("build"))?;
+    fs::copy(
+        &buildpack_binaries.buildpack_target_binary_path,
+        bin_path.join("build"),
+    )?;
+
     create_file_symlink("build", bin_path.join("detect"))?;
 
+    let additional_binaries_dir = destination_path
+        .as_ref()
+        .join(".libcnb-cargo")
```

Comment:
Woops, good catch! I will fix the wrong one. :) 

I wanted to have a prefix for these implicitly added files so that they'll never clash with something explicit. I was thinking about putting them in `bin` at first, but what happens if a user calls their binary `detect`? Or `env` which might be used by the CNB spec in the future?

By putting them in a `.libcnb-cargo` directory, we know these files from from `libcnb-cargo` and the change of clashing with something else is very low. I can see us using the directory for other things in the the future too. For example, we could put some build metadata there to aid with debugging.

Users, at least for the exec.d case will not have to construct this path by hand, a macro will be supplied that can be used to get a `PathBuf` to any Crate binary.

I don't see a problem with this deep path and would vote to keep it as-is. Thoughts?

Reply:
@edmorley and I discussed this on Slack and decided to keep the path as-is.